company.el

```diff
@@ -1448,6 +1448,55 @@ end of the match."
                                 candidate
                                 selected))
 
+(defvar company-unicode-icons-mapping
+  '((array . "Α")
+    (boolean . "β")
+    (class . "γ")
+    (color . "Δ")
+    (constant . "ε")
+    (enum-member . "ζ")
+    (enum . "Ζ")
+    (event . "η")
+    (field . "θ")
+    (file . "Ɩ")
+    (folder . "⍳")
+    (interface . "ϰ")
+    (key . "μ")
+    (keyword . "ν")
+    (method . "λ")
+    (function . "ƒ")
+    (misc . "ξ")
+    (module . "Ο")
+    (numeric . "π")
+    (operator . "⊙")
+    (parameter . "ρ")
+    (property . "σ")
+    (ruler . "τ")
+    (snippet . "υ")
+    (string . "φ")
+    (struct . "Χ")
+    (variable . "ѱ")))
+
+(defun company-unicode-icons-margin (candidate selected)
+  "Margin function which returns unicode icons."
+  (when-let ((candidate candidate)
+             (kind (company-call-backend 'kind candidate))
+             (icon (alist-get kind company-unicode-icons-mapping)))
+    (propertize icon
+                'face (if selected
+                          'company-tooltip-selection
+                        'company-tooltip))))
+
+(defun company-unicode-or-vscode-icons-margin (candidate selected)
+  "Margin function which picks from vscodes icons or unicode icons
+based on `display-graphic-p'."
+  (if (display-graphic-p)
+      ;; Default to dark because who in their right mind uses light 😜
```

Comment:
Just the users who care about ergonomics :stuck_out_tongue:

Reply:
Yeah, I think there are only 2 possible values, and it doesn't seem to ever return `nil`.

The dark icons might be the more reasonable fallback in the unlikely case it ever does because they're readable on while background. Not sure if the other set is as readable on black background.

Reply:
Done.